### ISSUE TITLE
spack find currently doesn't show packages which were specified as already installed in packages.yaml

### ISSUE THREAD

**Issue by krafczyk** [LLNL/spack#2191]
I've specified several implementations of mpi which are currently installed on my local computing cluster in packages.yaml.

When I run `spack find` it only lists packages which were explicitly built by spack itself, and doesn't mention these packages.

I think it would be good to also mention these packages or provide some kind of option to spack find to mention them as it helps verify that spack knows about these modules.

**Comment by alalazo** [comment]
That's because external packages are not registered in the DB. See #1066

**Comment by pramodk** [comment]
@alalazo : I have packages specified as an external which doesn't appear in `spack find`. But after `"some"` other dependent package installations, I have seen these packages appearing in the list. Not sure where is the inconsistency. For example, I never built `python` and specified as an `external` but I see now in the `spack find` (and install directory doesn't exist).

```bash
± |master {3} ?:9 ✗| → spack find -p python@2.6.6
==> 2 installed packages.

-- linux-redhat6-x86_64 / gcc@4.4.7 -----------------------------
    python@2.6.6  /mypath/spack/opt/spack/linux-redhat6-x86_64/gcc-4.4.7/python-2.6.6-z27rbmwpomzblrseotondvfq556w5tzw

-- linux-redhat6-x86_64 / pgi@16.5-0 ----------------------------
    python@2.6.6  /mypath/spack/opt/spack/linux-redhat6-x86_64/pgi-16.5-0/python-2.6.6-kdvmstycgmmekuoc6ifdo7grat4fr7jm

± |master {3} ?:9 ✗| → ls /mypath/spack/opt/spack/linux-redhat6-x86_64/gcc-4.4.7/python-2.6.6-z27rbmwpomzblrseotondvfq556w5tzw
ls: cannot access /mypath/spack/opt/spack/linux-redhat6-x86_64/gcc-4.4.7/python-2.6.6-z27rbmwpomzblrseotondvfq556w5tzw: No such file or directory
```

**Comment by alalazo** [comment]
@krafczyk @pramodk I was wondering if #1167 may solve the issue here...

**Comment by pramodk** [comment]
As #1167 is merged, we can close this issue. @krafczyk ?

**Comment by krafczyk** [comment]
@pramodk @alalazo Yes, this issue is resolved with #1167. I forgot to reply here.

**Comment by alalazo** [comment]
Yeah, now we have only 526 issues to go! :smile:
